### PR TITLE
Allow pyarrow >2.0.0

### LINKED ISSUE #6754: Unpin pyarrow in CI
We've temporarily pinned pyarrow<2 in the CI environments as there were some changes that broke our tests. We should test against pyarrow 2.

---

https://github.com/dask/dask/runs/1281335438?check_suite_focus=true#step:6:211

```
================================== FAILURES ===================================
_________________________ test_get_pyarrow_filesystem _________________________

    def test_get_pyarrow_filesystem():
        from fsspec.implementations.local import LocalFileSystem
    
        pa = pytest.importorskip("pyarrow")
    
        fs = LocalFileSystem()
>       assert isinstance(fs, pa.filesystem.FileSystem)
E       AssertionError: assert False
E        +  where False = isinstance(<fsspec.implementations.local.LocalFileSystem object at 0x00000224F12C08C8>, <class 'pyarrow.filesystem.FileSystem'>)
E        +    where <class 'pyarrow.filesystem.FileSystem'> = <module 'pyarrow.filesystem' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\filesystem.py'>.FileSystem
E        +      where <module 'pyarrow.filesystem' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\filesystem.py'> = <module 'pyarrow' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\__init__.py'>.filesystem

dask\bytes\tests\test_local.py:366: AssertionError
___________________________ test_get_pyarrow_fs_s3 ____________________________

s3 = <s3fs.core.S3FileSystem object at 0x00000224805B9BC8>

    def test_get_pyarrow_fs_s3(s3):
        pa = pytest.importorskip("pyarrow")
        fs = DaskS3FileSystem(anon=True)
>       assert isinstance(fs, pa.filesystem.FileSystem)
E       AssertionError: assert False
E        +  where False = isinstance(<s3fs.core.S3FileSystem object at 0x0000022482EA3BC8>, <class 'pyarrow.filesystem.FileSystem'>)
E        +    where <class 'pyarrow.filesystem.FileSystem'> = <module 'pyarrow.filesystem' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\filesystem.py'>.FileSystem
E        +      where <module 'pyarrow.filesystem' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\filesystem.py'> = <module 'pyarrow' from 'C:\\Miniconda3\\envs\\test-environment\\lib\\site-packages\\pyarrow\\__init__.py'>.filesystem

```

Some additional failures like ` test_append_with_partition[pyarrow] ` further down I think.

This is perhaps related to pyarrow 2.0 being released yesterday?

### PR DESCRIPTION
**Pull request by rjzamora**
Closes #6754 - Fix for pyarrow 2.0.0 errors will be fixed in the next release, so we will need to avoid 2.0.0

- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
Thanks @rjzamora! Glad to hear the upcoming `pyarrow` release has fixes : ) 

Could you update the `pyarrow` version constraint over in `dask/continuous_integration/hdfs/install.sh`?

File: continuous_integration/environment-3.6.yaml [hunk 1]
```diff
@@ -22,9 +22,7 @@ dependencies:
   - zarr
   - tiledb-py
   - sqlalchemy
-  # Remove <2 version constraint once we've added pyarrow 2.0 compatibility
-  # xref https://github.com/dask/dask/issues/6754
-  - pyarrow>=0.14.0,<2
+  - pyarrow>=0.14.0,<2|>2
```
Comment (rjzamora) [inline continuous_integration/environment-3.6.yaml:25]: @jrbourbeau (sorry for the basic question): Do you know if `>2` includes `2.1.0`, or does it exclude all `2.*`?

Comment (jrbourbeau) [reply continuous_integration/environment-3.6.yaml:25]: I think it should include `2.1.0`. Locally I created a conda environment with `conda create -n test python=3.8 "dask>2"` which installed `dask=2.30.0`

Comment (jrbourbeau) [reply continuous_integration/environment-3.6.yaml:25]: Though maybe something like `!=2.0.0` would be more clear if there's just a single version we want to exclude

**Review by jrbourbeau — APPROVED** [review]
Thanks @rjzamora!

File: continuous_integration/environment-3.8-dev.yaml [hunk 2]
```diff
@@ -19,9 +19,7 @@ dependencies:
   - zarr
   - tiledb-py
   - sqlalchemy
-  # Remove <2 version constraint once we've added pyarrow 2.0 compatibility
-  # xref https://github.com/dask/dask/issues/6754
-  - pyarrow>=0.14.0,<2
+  - pyarrow>=0.14.0,!=2.0.0
```
Comment (jorisvandenbossche) [inline continuous_integration/environment-3.8-dev.yaml:22]: Alternative could be to (temporarily) install the nightly wheel in this dev environment? (https://arrow.apache.org/docs/python/install.html#installing-nightly-packages) 
Then we would already have coverage for the new parquet dataset engine with pyarrow > 2.0 as well

Comment (jrbourbeau) [reply continuous_integration/environment-3.8-dev.yaml:22]: That's a good idea. I'm trying out CI with the nightly version of `pyarrow` over in https://github.com/dask/dask/pull/6936

Comment (jorisvandenbossche) [reply continuous_integration/environment-3.8-dev.yaml:22]: Another option could also be to skip the specific failing tests for pyrrow 2.0.0, but still install it in some CI environments (then at least we test the other (passing) tests with pyarrow 2.0)

**Pull request merged**